server/default.nix

```diff
@@ -0,0 +1,112 @@
+{ mkDerivation, aeson, aeson-casing, aeson-ordered, aeson-pretty
```

Comment:
Any reason not to use `callCabal2nix` for these?

If it's just for the version, you can override that later. I think this will work:

```nix
(callCabal2nix "graphql-engine" ./server {}).overrideAttrs(oldAttrs: {
  version = builtins.readFile ./server/CURRENT_VERSION;
})

Reply:
`callCabal2nix` is an anti-pattern in the flake/nixpkgs world, as it creates evaluations at runtime.
The computation of these files, while being more convenient, are done at eval time. 
https://github.com/input-output-hk/haskell.nix uses this approach extensively, I tried using that on a different branch in the past days and the IFD-less version of this branch is way more snappier to eval/build with. 
IMO for now it's worth to do this outside of Nix evalution, maybe automate it with a Makefile/CI script.

Reply:
Further reading:
https://github.com/input-output-hk/haskell.nix/blob/master/docs/tutorials/materialization.md

Reply:
Very helpful, thank you!

We'd need to add some CI to make sure this stays in sync with however it's generated, as well as include instructions on how to redo it.

Reply:
> We'd need to add some CI to make sure this stays in sync with however it's generated, as well as include instructions on how to redo it.

You can include a pre-commit hook [that runs cabal2nix](https://github.com/cachix/pre-commit-hooks.nix/blob/bd38df3d508dfcdff52cd243d297f218ed2257bf/modules/hooks.nix#L1047-L1053).

Note that devs don't have to install the hooks themselves, but only have to run `pre-commit run` after staging the changes. Else, `nix flake check` should be expected to fail.